cmd/vulcan-build-images/main.go

```diff
@@ -436,9 +448,139 @@ func forceRun(imagePath string) error {
 			env = append(env, fmt.Sprintf("%s=%s", k, v))
 		}
 	}
+
 	return util.RunCheckImage(imageName, env)
 }
 
+func forceRunReport(imagePath string, reportPath string) error {
+	imageName, err := forceBuild(imagePath)
+	if err != nil {
+		return err
+	}
+	var env []string
+	// Setup container env by reading, the local.toml file of the check.
+	// The file must exists.
+	cpath := path.Join(imagePath, "local.toml")
+	if _, err := os.Stat(cpath); err != nil {
+		return err
+	}
+	c, err := sdkconfig.LoadConfigFromFile(cpath)
+	if err != nil {
+		return err
+	}
+
+	allowPrivateIPs := true
+	if c.AllowPrivateIPs != nil {
+		allowPrivateIPs = *c.AllowPrivateIPs
+	}
+	if c.Log.LogLevel != "" {
+		env = append(env, "VULCAN_CHECK_LOG_LVL="+c.Log.LogLevel)
+	}
+	if c.Check.CheckID == "" {
+		c.Check.CheckID = uuid.New().String()
+	}
+
+	// Create a queue to store the check messages in memory.
+	q, err := queue.NewSimpleMQClientServer()
+	if err != nil {
+		return err
+	}
+	var qdone = make(chan error)
+	go func() {
+		err := q.Start()
+		qdone <- err
+	}()
+	err = q.WaitStart(time.Duration(5) * time.Second)
+	if err != nil {
+		return err
+	}
+	var (
+		agentAddr string
+		host      bool
+	)
+	// Only mac os and linux are supported.
+	if runtime.GOOS == "linux" {
+		agentAddr = getAgentAddr(q.Addr)
+		host = true
+	} else {
+		agentAddr = queueAddr(q.Addr)
+	}
+	// NOTE: the name of the env vars should be read from public constants of the sdk.
+	env = append(env, "VULCAN_CHECK_TARGET="+c.Check.Target)
+	env = append(env, "VULCAN_CHECK_OPTIONS="+c.Check.Opts)
+	env = append(env, "VULCAN_ALLOW_PRIVATE_IPS="+strconv.FormatBool(allowPrivateIPs))
+	env = append(env, fmt.Sprintf("%s=%s", "VULCAN_AGENT_ADDRESS", agentAddr))
+	env = append(env, fmt.Sprintf("%s=%s", "VULCAN_CHECK_ID", c.Check.CheckID))
+
+	for k, v := range c.RequiredVars {
+		env = append(env, fmt.Sprintf("%s=%s", k, v))
+	}
+	fmt.Printf("env passed to docker %+v", env)
+	err = util.RunCheckReportImage(imageName, env, c.Check.Target, host)
+	if err != nil {
+		cerr := closeQueue(&q, qdone)
```

Comment:
Won't be better a defer closeQueue?

Reply:
Yep, it could be used, the think here is that I wanted to explicitly handle the posible error when closing the, that can happen when also other are triggered. For that I would need to somehow access the previous error, like in the line below:
```go
err = fmt.Errorf("error: %s, and error closing the queue %w", err.Error(), cerr)
```
 This can be done, of course, using a defer clause, something like this:
```go
defer func() {
   if err != nil {
      fmt.Printf("error got %+v", err)
   }
  }() 
```
But it's not usually a good practice to have complicated defer functions because they are hard to read and follow.

Reply:
So I preferred to take the trade-of of repeating a little bit the code in exchange of making the code more explicit.

Reply:
Makes sense, thanks for the explanation!